### ISSUE TITLE
Emoji History Is Formed Only from Explicitly Selected Emojis

### ISSUE THREAD

**Issue by stanislawru** [telegramdesktop/tdesktop#16926]
### Steps to reproduce

1. Open any chat.
2. **Visually select** any emoji from the emoji panel and then send it to that chat, better a few times.
3. See that emoji in the list of recent emojis.
4. **Type** any **other** emoji **manually** as :emojiname: or paste it from the clipboard and then send it to the chat, better a few times.
5. See that second emoji has **not** appeared in the list of recent emojis.

### Expected behaviour

The emoji history must be formed **not only** from the emojis **visually selected** from the emoji panel but **also from** those that were **typed manually** or **pasted from the clipboard**.

### Actual behaviour

Emojis that were **typed manually** or **pasted from the clipboard** do not appear in the list of recent emojis.

### Operating system

Windows 7, 8.1 and 10

### Version of Telegram Desktop

3.0.1

### Installation source

Static binary from official website

**Comment by ilya-fedin** [comment]
> or **pasted from the clipboard**

I don't think it's a good idea. Imagine you pasted a big text with a lot of emojis, boom, all of these emojis will be in your list of recent emojis, removing emojis you actually using regularly.

**Comment by stanislawru** [comment]
OK, let's limit fix to: emojis that were **typed manually** as :emojiname: **should appear** in the list of recent emojis. This works well in the Android app.

**Comment by stanislawru** [comment]
_“We therefore assume that the user has lost interest or resolved the problem on their own”_, – really? :open_mouth: I am not a developer, so cannot fix the issue myself. And I consider unfair to close the issue just beacuse there is no developer to fix it. :disappointed:

**Comment by stanislawru** [comment]
_“We therefore assume that the user has lost interest or resolved the problem on their own”_, – really? :open_mouth: I am not a developer, so cannot fix the issue myself. And I consider unfair to close the issue just beacuse there is no developer to fix it. :disappointed:

**Comment by ilya-fedin** [comment]
Yeah, really. A lot of people losing interest in issues after some time. If you're not, relax and just answer the bot for it to not to close the issue.

**Comment by stanislawru** [comment]
_“We therefore assume that the user has lost interest or resolved the problem on their own”_, – really? 😮 I am not a developer, so cannot fix the issue myself. And I consider unfair to close the issue just because there is no developer to fix it. 😞

**Comment by stanislawru** [comment]
_“We therefore assume that the user has lost interest or resolved the problem on their own”_, – really? 😮 I am not a developer, so cannot fix the issue myself. And I consider unfair to close the issue just because there is no developer to fix it. 😞

**Comment by john-preston** [comment]
@stanislawru 66

**Comment by stanislawru** [comment]
_“We therefore assume that the user has lost interest or resolved the problem on their own”_, – really? 😮 I am not a developer, so cannot fix the issue myself. And I consider unfair to close the issue just because there is no developer to fix it. 😞

**Comment by stanislawru** [comment]
_“We therefore assume that the user has lost interest or resolved the problem on their own”_, – really? 😮 I am not a developer, so cannot fix the issue myself. And I consider unfair to close the issue just because there is no developer to fix it. 😞

**Comment by stanislawru** [comment]
_“We therefore assume that the user has lost interest or resolved the problem on their own”_, – really? 😮 I am not a developer, so cannot fix the issue myself. And I consider unfair to close the issue just because there is no developer to fix it. 😞

**Comment by stanislawru** [comment]
_“We therefore assume that the user has lost interest or resolved the problem on their own”_, – really? 😮 I am not a developer, so cannot fix the issue myself. And I consider unfair to close the issue just because there is no developer to fix it. 😞

**Comment by stanislawru** [comment]
_“We therefore assume that the user has lost interest or resolved the problem on their own”_, – really? 😮 I am not a developer, so cannot fix the issue myself. And I consider unfair to close the issue just because there is no developer to fix it. 😞

**Comment by stanislawru** [comment]
_“We therefore assume that the user has lost interest or resolved the problem on their own”_, – really? 😮 I am not a developer, so cannot fix the issue myself. And I consider unfair to close the issue just because there is no developer to fix it. 😞
